src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/TestHelper.java

```diff
@@ -149,15 +152,23 @@ public String getRandomFilename(String suffix) {
             if (metadataFile != null) {
                 String metadataTemplate =
                     Resources.toString(Resources.getResource(metadataFile), StandardCharsets.UTF_8);
-                String metadata = metadataTemplate.replace("$$zip_file_name$$", zipFilename);
+                String metadata = metadataTemplate
+                    .replace("$$zip_file_name$$", zipFilename)
+                    .replace("$$dcn1$$", generateDcnNumber())
+                    .replace("$$dcn2$$", generateDcnNumber());
```

Comment:
This method accepts a dynamic list of file names, but then assumes there are exactly two of them. If that's always the case, maybe we could align the parameters in the method's signature with these replacements?

Reply:
previous suggestion to use `reduce` should solve any doubts (or so i hope). wasn't thinking in such way for a while tbh hence few raw implementation attempts as you inspect

Reply:
Oh wait. I missed the fact that this is a completely new exception. In this case it has to implement that interface (which is nothing but a marker) and all should be fine.

Reply:
Why I put the above comment here is a mistery even to me